syntax/json.vim

```diff
@@ -16,30 +16,38 @@ if !exists("main_syntax")
   let main_syntax = 'json'
 endif
 
+" parse options and set vars
+if has('conceal') && g:vim_json_syntax_conceal == 1
+   let conceal = " concealends"
+else
+   let conceal = ""
+endif
+
+if g:vim_json_color_quotes == 1
+	let keyword_quote_match_group = "jsonKeywordQuote"
+	let string_quote_match_group  = "jsonStringQuote"
+else
+	let keyword_quote_match_group = "jsonQuote"
+	let string_quote_match_group  = "jsonQuote"
+endif
+
+
 syntax match   jsonNoise           /\%(:\|,\)/
 
 " NOTE that for the concealing to work your conceallevel should be set to 2
 
 " Syntax: Strings
 " Separated into a match and region because a region by itself is always greedy
 syn match  jsonStringMatch /"\([^"]\|\\\"\)\+"\ze[[:blank:]\r\n]*[,}\]]/ contains=jsonString
-if has('conceal') && g:vim_json_syntax_conceal == 1
-	syn region  jsonString oneline matchgroup=jsonQuote start=/"/  skip=/\\\\\|\\"/  end=/"/ concealends contains=jsonEscape contained
-else
-	syn region  jsonString oneline matchgroup=jsonQuote start=/"/  skip=/\\\\\|\\"/  end=/"/ contains=jsonEscape contained
-endif
+execute 'syn region  jsonString oneline matchgroup=' . string_quote_match_group . ' start=/"/  skip=/\\\\\|\\"/  end=/"/ contains=jsonEscape contained' . conceal
```

Comment:
instead of running different syn commands based on user options, we'll set some vars and build the desired syn command from those